src/gc/impl/precise/gc.d

```diff
@@ -1930,38 +2150,51 @@ struct Gcx
         void grow()
         {
             enum initSize = 64 * 1024; // Windows VirtualAlloc granularity
-            immutable ncap = _cap ? 2 * _cap : initSize / Range.sizeof;
-            auto p = cast(Range*)os_mem_map(ncap * Range.sizeof);
+            immutable ncap = _cap ? 2 * _cap : initSize / T.sizeof;
+            auto p = cast(T*)os_mem_map(ncap * T.sizeof);
             if (p is null) onOutOfMemoryErrorNoGC();
             if (_p !is null)
             {
                 p[0 .. _length] = _p[0 .. _length];
-                os_mem_unmap(_p, _cap * Range.sizeof);
+                os_mem_unmap(_p, _cap * T.sizeof);
             }
             _p = p;
             _cap = ncap;
         }
 
         size_t _length;
-        Range* _p;
+        T* _p;
         size_t _cap;
     }
 
-    ToScanStack toscan;
+    ToScanStack!(Range) toscanConservative;
+    ToScanStack!(ScanRange) toscanPrecise;
 
     /**
      * Search a range of memory values and mark any pointers into the GC pool.
      */
-    void mark(void *pbot, void *ptop) nothrow
+    void mark(bool precise)(void *pbot, void *ptop) nothrow
     {
+        static if(precise)
+            alias toscan = toscanPrecise;
+        else
+            alias toscan = toscanConservative;
+
```

Comment:
Defining an alias for the range type here avoids a number of `static if (precise)` below.